client/lib/plans/features-list.tsx

```diff
@@ -1769,6 +1780,22 @@ export const FEATURES_LIST: FeatureList = {
 		getAlternativeTitle: () => '0%',
 		getFeatureGroup: () => FEATURE_GROUP_PAYMENT_TRANSACTION_FEES,
 	},
+	[ FEATURE_50GB_STORAGE_ADD_ON ]: {
+		getSlug: () => PRODUCT_1GB_SPACE,
+		getQuantity: () => 50,
+		getTitle: () => i18n.translate( '100 GB' ),
```

Comment:
Similar - is `100 GB` correct here? (also for the other cases below)

Reply:
Yes it is.

From what I can tell, `getTitle` is used to render strings in the features grid table. The add on string we use is `100 GB` because what we render in the drop down it is the 50GB default + the 50GB add-on for a total of 100GB

Reply:
Wow this is actually weird! that we have a feature named "50GB_STORAGE" but its title says "100 GB" - because it's used with a plan on a default of 50 GBs (os as the second item in a drop-down that starts with 50 and renders multiple of 50s)?? This definitely falls on the ambiguous side for me.

> The add on string we use is 100 GB because what we render in the drop down is the 50GB default + the 50GB add-on for a total storage of 100GB

This sounds very much like a UI or consuming-end concern. A `getTitle` property should leave no ambiguity or let context creep in like that. I'd flag this as "typo" every time I'd come across really. So being "intentional" alone doesn't justify this. 

Without thinking deeper about this, here are a couple of options that I'd consider:

1. Make the UI/consuming end a little more complex by doing the calculations on that end (that it should show 100GB instead of 50GB that the add-on refers to). In this solution, `getTitle` is statically pointing to "50 GB", irrespective of use, it leaves no ambiguity in the object definition itself. The UI can render whatever it wants, but anything else consuming this object should trust the title to be precise and relevant.
2. Pass an optional "baseQuantity" variable (or something) to the getter to do the necessary addition and return the resulting string. It should also be clear what the title refers to, but I would consider renaming it to something else (which starts to feel again like a UI/consuming concern creeping into this).

Also, the `getSlug` property feels a little generic to point to a product that is consumed in multiples of `getQuantity`. Maybe refactor that to something like `getUnitProductSlug` or something. It needs to be clarified. Someone looking at these properties should be able to understand right off the bat what they are referring to. This is far from being the case right now AFAIC.

Reply:
>... - so maybe not

Thinking whether the actual storage dropdown options could be refactored and reused across places instead in that case (instead of complicating the feature object to expose complex titles)

^ is this something we are trying to achieve? Use in plans UI _and_ checkout? Give me a bit more context pls to understand how we ended up with this

Reply:
> Make the UI/consuming end a little more complex by doing the calculations on that end (that it should show 100GB instead of 50GB that the add-on refers to). In this solution, getTitle is statically pointing to "50 GB", irrespective of use, it leaves no ambiguity in the object definition itself. The UI can render whatever it wants, but anything else consuming this object should trust the title to be precise and relevant.

This is what was happening originally. There was a helper method being implemented in the presentation layer which is where I was originally modifying the displayed string. 

> ^ is this something we are trying to achieve? Use in plans UI and checkout? Give me a bit more context pls to understand how we ended up with this

Correct. I was moving in that direction with this change

Correct. There is an identical dropdown in the checkout

Reply:
Slugs resolved in [d2a2086](https://github.com/Automattic/wp-calypso/pull/79041/commits/d2a20867da3527e9bd194567aa10911a4b28555a)
Merge state
Add more commits by pushing to the [add/plans-2023-upgradeable-storage](https://github.com/Automattic/wp-calypso/tree/add/plans-2023-upgradeable-storage) branch on [Automattic/wp-calypso](https://github.com/Automattic/wp-calypso).

This branch has not been deployed
No deployments
Changes requested
1 review requesting changes 
1 change requested
2 pending reviewers
Some checks haven’t completed yet
1 queued and 2 expected checks
@github-actions
Calculate ICFY stats / Build ICFY stats (push) Queued — Waiting to run this check...
[Details](https://github.com/Automattic/wp-calypso/actions/runs/5720038514/job/15499215868?pr=79041)
Code style (Web app) Expected — Waiting for status to be reported
Required
Unit tests (Web app) Expected — Waiting for status to be reported
Required
Required statuses must pass before merging
All required [statuses](https://docs.github.com/rest/commits/statuses) and check runs on this pull request must run successfully to enable automatic merging.
 You can also [open this in GitHub Desktop](x-github-client://openRepo/https://github.com/Automattic/wp-calypso?branch=add%2Fplans-2023-upgradeable-storage) or view .
@jeyip

 
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/Automattic/wp-calypso/blob/f711e4b98675a986d37a80811029edf5abc2c4e7/docs/CONTRIBUTING.md), [security policy](https://github.com/Automattic/wp-calypso/blob/f711e4b98675a986d37a80811029edf5abc2c4e7/SECURITY.md), and [code of conduct](https://github.com/Automattic/wp-calypso/blob/f711e4b98675a986d37a80811029edf5abc2c4e7/CODE-OF-CONDUCT.md).
 ProTip! Add [.patch](https://github.com/Automattic/wp-calypso/pull/79041.patch) or [.diff](https://github.com/Automattic/wp-calypso/pull/79041.diff) to the end of URLs for Git’s plaintext views.
Reviewers
@[chriskmnds](https://github.com/chriskmnds)
chriskmnds
@[southp](https://github.com/southp)
southp
@[aneeshd16](https://github.com/aneeshd16)
aneeshd16
@[gmovr](https://github.com/gmovr)
gmovr
Still in progress?
Assignees
@[jeyip](https://github.com/jeyip)
jeyip
Labels
[2023 Pricing Page ⚡](https://github.com/Automattic/wp-calypso/labels/2023%20Pricing%20Page%20%E2%9A%A1)
[[Status] Needs Review](https://github.com/Automattic/wp-calypso/labels/%5BStatus%5D%20Needs%20Review)
Projects
[ MarTech Task Board](https://github.com/orgs/Automattic/projects/388)

Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

None yet

Notifications
Customize
You’re receiving notifications because you were mentioned.
5 participants
@jeyip
@matticbot
@jdc91
@chriskmnds
@southp